packages/appium/lib/cli/parser.js

```diff
@@ -242,15 +250,18 @@ class ArgParser {
         },
       ];
 
-      for (const {command, args, help} of parserSpecs) {
-        const parser = extSubParsers.add_parser(command, {help});
+      for (const {command, args, help, aliases} of parserSpecs) {
+        const parser = extSubParsers.add_parser(command, {help, aliases: aliases ?? []});
 
         ArgParser._patchExit(parser);
 
         for (const [flagsOrNames, opts] of args) {
           // add_argument mutates params so make sure to send in copies instead
-          // @ts-ignore
-          parser.add_argument(...flagsOrNames, {...opts});
+          if (flagsOrNames.length === 2) {
```

Comment:
could you please explain this magic?

Reply:
If there was type inference available in GH's review tool (maybe this is some addon app that does this), then you probably wouldn't have asked. 😄 

`args` is of type `ArgumentDefinitions`, which is defined [here](https://github.com/appium/appium/pull/18479/files#diff-ee3a38dac1a7e9aeb4ade3c6bd97e6bcabf8690782f2666621dedcab01008a04R297).  This is a tuple, where the type of the first value (`flagsOrNames`) is `[string] | [string, string]`.  That is _also_ a tuple (an array w/ a fixed number of items) with either one or two items.  

I noticed when looking at `argparser`'s implementation that it only allows a single alias for any given arg.  So we can either give it the option name (`[string]`), or the option name and an alias (`[string, string]`).

Unfortunately, that conflicts with our config schema that says a server option or CLI argument can have an arbitrary number of aliases.  I'm punting on it for now, but I've noted the problem in comment(s).

Reply:
Of note, TS treats function parameters as tuples.  For any function `f`, you can get a tuple of its parameters via `Parameters<typeof f>`.  `argparser`'s `add_argument()` function looks like this:

```ts
add_argument(arg: string, options?: ArgumentOptions): void;
add_argument(arg1: string, arg2: string, options?: ArgumentOptions): void;
```

This is why we're only allowed the name and a single alias.